src/cbor/fallback_decoder.rs

```diff
@@ -19,21 +22,40 @@ struct FDRequest {
     response_tx: oneshot::Sender<Result<serde_json::Value, String>>,
 }
 
-const CHILD_EXE_NAME: &str = "testgen-hs";
-
 impl FallbackDecoder {
     /// Starts a new child process.
-    pub fn spawn() -> Self {
+    pub fn spawn() -> Result<Self, AppError> {
+        let mut search_roots = vec![
+            PathBuf::from("/usr/bin"),
+            PathBuf::from("/bin"),
+            PathBuf::from("target/testgen-hs/extracted/testgen-hs"),
+            PathBuf::from("/app/testgen-hs"),
+        ];
+
+        if let Ok(path_var) = env::var("PATH") {
+            for dir in path_var.split(MAIN_SEPARATOR) {
```

Comment:
There's also [`std::env::split_paths`](https://doc.rust-lang.org/std/env/fn.split_paths.html):

```rust
            for dir in env::split_paths(&path_var) …
```

But either way it works :)